File: app/src/main/java/com/gtm/archcomponents/notes/ui/adapter/NotesAdapter.java
package com.gtm.archcomponents.notes.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.gtm.archcomponents.R;
import com.gtm.archcomponents.notes.room.Notes;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.subjects.PublishSubject;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.NotesViewHolder> {

    private List<Notes> notes;
    private final LayoutInflater mInflater;
    private View mView;
    private PublishSubject<Integer> position = PublishSubject.create();
    private int selectedPosition = -1;

    class NotesViewHolder extends RecyclerView.ViewHolder {

        TextView notesTitle;
        TextView notesBody;
        TextView notesTime;
        ImageView noteSelection;

        public NotesViewHolder(@NonNull View itemView) {
            super(itemView);
            notesTitle = itemView.findViewById(R.id.tv_note_title);
            notesBody = itemView.findViewById(R.id.tv_note_body);
            notesTime = itemView.findViewById(R.id.id_tv_time);
            noteSelection = itemView.findViewById(R.id.id_im_selection);


            itemView.setOnLongClickListener(v -> {
                        if (selectedPosition == getAdapterPosition()) {
                            selectedPosition = -1;
                            notifyDataSetChanged();
                        } else {
                            selectedPosition = getAdapterPosition();
                            position.onNext(getAdapterPosition());
                            notifyDataSetChanged();
                        }
                        return true;
                    }
            );
        }
    }


    public NotesAdapter(Context context) {
        this.mInflater = LayoutInflater.from(context);
    }


    public Observable<Integer> getLongClickedItemPosition() {

        return position;
    }

    @NonNull
    @Override
    public NotesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        mView = this.mInflater.inflate(R.layout.list_item_note, parent, false);

        return new NotesViewHolder(mView);
    }

    @Override
    public void onBindViewHolder(@NonNull NotesViewHolder holder, int position) {

        if (selectedPosition == position) {
            holder.noteSelection.setVisibility(View.VISIBLE);
            selectedPosition = -1;
        } else {
            holder.noteSelection.setVisibility(View.GONE);
        }

        holder.notesBody.setText(notes.get(position).getNote());
        holder.notesTitle.setText(notes.get(position).getNoteTitle());
        holder.notesTime.setText(notes.get(position).getCreated_time());


    }

    @Override
    public int getItemCount() {
        return (notes == null) ? 0 : notes.size();
    }

    public void setNotes(List<Notes> notes) {
        this.notes = notes;
        notifyDataSetChanged();
    }


}

File: app/src/main/java/com/gtm/pustak/kotlin/ui/adapter/NotesAdapter.kt
package com.gtm.pustak.kotlin.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.gtm.pustak.R
import com.gtm.pustak.kotlin.Utils
import com.gtm.pustak.notes.room.Notes
import io.reactivex.Observable
import io.reactivex.subjects.PublishSubject

class NotesAdapter(context: Context) : RecyclerView.Adapter<NotesAdapter.NotesViewHolder>() {
    var selectedPosition: Int = -1
    var positionPublisher: PublishSubject<Int> = PublishSubject.create<Int>()
    private var layoutInflater: LayoutInflater = LayoutInflater.from(context)
    private var notesList: List<Notes>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotesViewHolder {
        return NotesViewHolder(layoutInflater.inflate(R.layout.list_item_note, parent, false))
    }

    override fun getItemCount(): Int {
        return if (notesList == null) 0 else (notesList!!.size)

    }

    override fun onBindViewHolder(holder: NotesViewHolder, position: Int) {
        if (selectedPosition == position) {
            holder.noteSelection.visibility = View.VISIBLE
            selectedPosition = -1
        } else {
            holder.noteSelection.visibility = View.GONE
        }

        holder.notesBody.text = notesList!!.get(position).note
        holder.notesTitle.text = (notesList!!.get(position).getNoteTitle())

        var createdTime = Utils.getTimeWithoutSeconds(notesList!!.get(position).created_time)

        holder.notesTime.text = (createdTime)
    }

    fun setNotes(notes: List<Notes>) {
        this.notesList = notes
        notifyDataSetChanged()

    }


    fun getlongClickedItemPosition(): Observable<Int> {
        return positionPublisher
    }

    inner class NotesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var notesTitle: TextView
        var notesBody: TextView
        var notesTime: TextView
        var noteSelection: ImageView

        init {
            notesTitle = itemView.findViewById(R.id.tv_note_title);
            notesBody = itemView.findViewById(R.id.tv_note_body)
            notesTime = itemView.findViewById(R.id.id_tv_time)
            noteSelection = itemView.findViewById(R.id.id_im_selection)
            itemView.setOnLongClickListener { v ->

                if (selectedPosition == adapterPosition) {
                    selectedPosition = -1
                    notifyDataSetChanged()
                } else {
                    selectedPosition = adapterPosition
                    positionPublisher.onNext(selectedPosition)
                    notifyDataSetChanged()
                }
                false
            }

        }


    }

}